api/services/BuildResolver.js

```diff
@@ -0,0 +1,73 @@
+const { Build, Site } = require('../models');
+const buildErrors = require('../responses/buildErrors');
+const GitHub = require('../services/GitHub');
+
+const rejectBuild = msg =>
+  Promise.reject({
+    status: 404,
+    message: msg,
+  });
+
+const getBranch = ({ user, site, branch }) =>
+  GitHub.getBranch(user, site.owner, site.repository, branch)
+  .then(branchInfo => ({
+    branch: branchInfo.name,
+    site: site.id,
+    commitSha: branchInfo.commit.sha,
+  }))
+  .catch(() => rejectBuild(buildErrors.BRANCH_NOT_FOUND));
+
+const getBuildById = (user, params) => {
+  const { buildId, siteId } = params;
+
+  return Build.findOne({
+    where: {
+      id: buildId,
+      site: siteId,
+    },
+  })
+  .then((model) => {
+    if (model) {
+      return model;
+    }
+
+    return rejectBuild(buildErrors.BUILD_NOT_FOUND);
+  });
+};
+
+const getBuildByBranch = (user, params) => {
+  const { siteId, sha, branch } = params;
+
+  return Site.findOne({
+    where: { id: siteId },
+    attributes: ['id', 'owner', 'repository'],
+    include: {
+      model: Build,
+      where: { branch },
+      required: false,
+    },
+  })
+  .then((model) => {
+    const site = model;
+    const build = site.Builds && site.Builds[0];
```

Comment:
Is the first `Build` guaranteed to be the one we want?

Reply:
Like maybe in the `include` config they should be ordered descending by `createdAt` to make sure we're getting the latest one at the `[0]` element?

Reply:
You might also be able to add `limit: 1` to the the `include` config for a slightly speedier database query.

Reply:
No you're right, when I shuffled things around from relying on the user, I didn't account for the fact that we could now get more than one build back.